src/main/java/com/commercetools/sync/services/impl/CustomObjectServiceImpl.java

```diff
@@ -0,0 +1,108 @@
+package com.commercetools.sync.services.impl;
+
+import com.commercetools.sync.commons.BaseSyncOptions;
+import com.commercetools.sync.commons.models.NonResolvedReferencesCustomObject;
+import com.commercetools.sync.services.CustomObjectService;
+import io.sphere.sdk.customobjects.CustomObject;
+import io.sphere.sdk.customobjects.CustomObjectDraft;
+import io.sphere.sdk.customobjects.commands.CustomObjectDeleteCommand;
+import io.sphere.sdk.customobjects.commands.CustomObjectUpsertCommand;
+import io.sphere.sdk.customobjects.queries.CustomObjectQuery;
+import io.sphere.sdk.queries.PagedResult;
+import io.sphere.sdk.queries.QueryPredicate;
+import org.apache.commons.lang3.StringUtils;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.stream.Collectors;
+
+import static java.lang.String.format;
+import static java.util.Collections.singleton;
+import static org.apache.commons.lang3.StringUtils.isBlank;
+
+public class CustomObjectServiceImpl
+        implements CustomObjectService {
+
+    private final BaseSyncOptions syncOptions;
+
+    private static final String CREATE_FAILED = "Failed to create draft with key: '%s'. Reason: %s";
+
+    public CustomObjectServiceImpl(final BaseSyncOptions baseSyncOptions) {
+        this.syncOptions = baseSyncOptions;
+    }
+
+    @Nonnull
+    @Override
+    public CompletionStage<Optional<CustomObject<NonResolvedReferencesCustomObject>>>
+        fetchCustomObject(@Nullable final String key) {
+
+        if (isBlank(key)) {
+            return CompletableFuture.completedFuture(Optional.empty());
+        }
+
+        final CustomObjectQuery<NonResolvedReferencesCustomObject> customObjectQuery = CustomObjectQuery
+                .of(NonResolvedReferencesCustomObject.class)
+                .withPredicates(buildCustomObjectKeysQueryPredicate(singleton(key)));
+
+        return syncOptions
+                .getCtpClient()
+                .execute(customObjectQuery)
+                .thenApply(PagedResult::head);
+    }
+
+    private QueryPredicate<CustomObject<NonResolvedReferencesCustomObject>> buildCustomObjectKeysQueryPredicate(
```

Comment:
Why do we need this method that takes set of keys? You are always calling this with `singleton(key)`!

Reply:
yes, this method seems to be unnecessary.

Reply:
2ed582ee415adbad4cb71228089cb5f21fd9568a